src/components/Tokens/TokenDetails/index.tsx

```diff
@@ -146,34 +145,35 @@ export default function TokenDetails({
   )
   useOnGlobalChainSwitch(navigateToTokenForChain)
 
-  const navigateToWidgetSelectedToken = useCallback(
-    (tokens: SwapTokens) => {
-      const newDefaultToken = tokens[Field.OUTPUT] ?? tokens.default
-      const address = newDefaultToken?.isNative ? NATIVE_CHAIN_ID : newDefaultToken?.address
+  const handleCurrencyChange = useCallback(
+    (tokens: Pick<SwapState, Field.INPUT | Field.OUTPUT>) => {
+      if (
+        addressesAreEquivalent(tokens[Field.INPUT]?.currencyId, address) ||
+        addressesAreEquivalent(tokens[Field.OUTPUT]?.currencyId, address)
+      ) {
+        return
+      }
+      const newDefaultTokenID = tokens[Field.OUTPUT]?.currencyId ?? tokens[Field.INPUT]?.currencyId
       startTokenTransition(() =>
         navigate(
           getTokenDetailsURL({
-            address,
+            address: newDefaultTokenID === 'ETH' ? 'NATIVE' : newDefaultTokenID,
```

Comment:
nit: should we create a constant file in constants for this? ETH/NATIVE? Looks like a follow-up task, b/c we're not using that anywhere. Just looks strange to be using strings.

Reply:
actually good point, we have a constant used elsewhere for the "NATIVE" string , and `getTokenDetailsURL` uses it as the fallback. i guess i should be using `null` here instead!